jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/component/Environment.java

```diff
@@ -32,22 +28,22 @@ public interface Environment extends Attributes
 
     static Collection<Environment> getAll()
     {
-        return Collections.unmodifiableCollection(Named.__environments.values());
+        return Collections.unmodifiableCollection(NamedEnvironment.__environments.values());
     }
     
     static Environment get(String name)
     {
-        return Named.__environments.get(name);
+        return NamedEnvironment.__environments.get(name);
     }
 
     static Environment ensure(String name)
     {
-        return Named.__environments.computeIfAbsent(name, Named::new);
+        return ensure(name, null);
     }
 
-    static Environment set(Environment environment)
+    static Environment ensure(String name, ClassLoader classLoader)
```

Comment:
This should be a `Supplier<ClassLoader>` argument, as ensure semantics should only construct a classloader if the environment doesn't already exist.

Reply:
If `EnvironmentBuilder.build()` is supposed to throw when an environment with such name already exists, there is no need to use a supplier for the classloader.

Reply:
one or the other